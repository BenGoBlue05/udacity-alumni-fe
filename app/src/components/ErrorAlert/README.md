## ErrorAlert Component
A component that shows alert notifications, given an input array of errors.

### Example

```js
<ErrorAlert onClose={this.handleClose} errors={myErrors} />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **errors**    | Array   |             | An array of errors with corresponding messages.
| **onClose**    | Function   |             | A function that is called when the close button is pressed.


### Other Information
Currently only showing as critical errors, although there are multiple types of errors.

See: http://grommet.github.io/docs/notification

The onClose callback will be called with an index, so the reducer needs to slice out the remaining errors.  See the reducer file in LoginContainer for reference.
