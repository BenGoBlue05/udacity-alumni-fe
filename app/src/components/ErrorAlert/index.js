import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Notification from 'grommet/components/notification';
import CloseIcon from 'grommet/components/icons/base/Close';
import Button from 'grommet/components/button';
import Section from 'grommet/components/Section';

const ErrorAlert = ({
  errors,
  onClose,
}) => (
  <Section className={styles.errorAlert}>
    {errors.length > 0 && errors.map((error, i) =>
      <div key={i} className={styles.error}>
        <div className="error-alert__closer">
          <Button
            plain
            onClick={() => onClose(i)}
            className={styles.closeButton}
            a11yTitle="Close Alert"
          >
            <CloseIcon a11yTitle="Close Alert" />
          </Button>
        </div>
        <Notification
          role="alert"
          style={{ paddingTop: 10 }}
          status="critical"
          a11yTitle="Submission Failed"
          size="small"
          message={error.message}
        />
      </div>
    )}
  </Section>
);

ErrorAlert.propTypes = {
  errors: PropTypes.array,
  onClose: PropTypes.func.isRequired,
};

export default cssModules(ErrorAlert, styles);
