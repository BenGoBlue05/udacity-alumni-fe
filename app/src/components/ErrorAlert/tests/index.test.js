import ErrorAlert from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<ErrorAlert />', () => {
  it('should render with default props', () => {
    const errors = [
      { message: 'A bad, bad error has occured :(' },
      { message: 'An even worse error has occured' },
    ];
    const wrapper = shallow(
      <ErrorAlert onClose={(e) => e} errors={errors} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
