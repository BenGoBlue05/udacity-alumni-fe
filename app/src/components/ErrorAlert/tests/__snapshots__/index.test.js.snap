exports[`<ErrorAlert /> should render with default props 1`] = `
<Section
  pad={
    Object {
      "vertical": "medium"
    }
  }>
  <div>
    <div
      className="error-alert__closer">
      <Button
        a11yTitle="Close Alert"
        onClick={[Function onClick]}
        plain={true}
        type="button">
        <Close
          a11yTitle="Close Alert"
          a11yTitleId="close-title"
          responsive={true} />
      </Button>
    </div>
    <Notification
      a11yTitle="Submission Failed"
      closer={false}
      flush={true}
      message="A bad, bad error has occured :("
      pad="medium"
      role="alert"
      size="small"
      status="critical"
      style={
        Object {
          "paddingTop": 10
        }
      } />
  </div>
  <div>
    <div
      className="error-alert__closer">
      <Button
        a11yTitle="Close Alert"
        onClick={[Function onClick]}
        plain={true}
        type="button">
        <Close
          a11yTitle="Close Alert"
          a11yTitleId="close-title"
          responsive={true} />
      </Button>
    </div>
    <Notification
      a11yTitle="Submission Failed"
      closer={false}
      flush={true}
      message="An even worse error has occured"
      pad="medium"
      role="alert"
      size="small"
      status="critical"
      style={
        Object {
          "paddingTop": 10
        }
      } />
  </div>
</Section>
`;
